core/modules/wmgr/python/dbMgr.py

```diff
@@ -0,0 +1,692 @@
+    return _validateId("Table", tblName)
+
+def _validateColumnName(columnName):
+    """ Validate column name """
+    return _validateId("Column", columnName)
+
+# list of database names that we want to ignore
+_specialDbs = set(['mysql', 'information_schema'])
+
+# pattern for CREATE TABLE
+_createTableRe = re.compile(r'^\s*CREATE\s+TABLE\s+(\w\.)?(\w+)', re.I)
+
+def _dbDict(dbName):
+    """ Make database instance dict out of db name """
+    return dict(name=dbName, uri=url_for('.dbDelete', dbName=dbName))
+
+def _tblDict(dbName, tblName):
+    """ Make table instance dict out of table name """
+    return dict(name=tblName, uri=url_for('.tableDelete', dbName=dbName, tblName=tblName))
+
+def _chunkDict(dbName, tblName, chunkId):
+    """ Make table instance dict out of table name """
+    uri = url_for('.chunkDelete', dbName=dbName, tblName=tblName, chunkId=chunkId)
+    return dict(chunkId=chunkId, uri=uri, chunkTable=False, overlapTable=False)
+
+def _getArgFlag(mdict, option, default=True):
+    """
+    Extracts specified argument from request arguments and converts it to boolean flag.
+    Argument is required ot have value, possible values are '0', '1', 'yes', 'no', 'true', 'false'.
+    """
+    value = mdict.get(option, None)
+    if value is None:
+        return default
+    value = value.strip().lower()
+    if value not in ('0', '1', 'yes', 'no', 'true', 'false'):
+        raise ExceptionResponse(400, "InvalidArgument",
+                                "Unexpected value of '%s' option: \"%s\"" % (option, value))
+    return value in ('1', 'yes', 'true')
+
+def _typeCode2Name(code):
+    """
+    Convert mysql type code to type name, returns None if ther is no mapping.
+    This is mysql-specific, it would be better to have technology-neutral way
+    to provide the same mapping.
+    """
+    for name in dir(FIELD_TYPE):
+        if getattr(FIELD_TYPE, name) == code:
+            return name
+    return None
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+dbService = Blueprint('dbService', __name__, template_folder='dbService')
+
+
+@dbService.errorhandler(db.DbException)
+def dbExceptionHandler(error):
+    """ All leaked DbException exceptions make 500 error """
+    return errorResponse(500, "db.DbException", str(error))
+
+@dbService.route('', methods=['GET'])
+def dbs():
+    """ Return the list of databases """
+
+    _log.debug('request: %s', request)
+    _log.debug('dbs() GET => get database list')
+
+    # use non-privileged account, may limit the list of databases returned.
+    dbConn = Config.instance().dbConn()
+    dbs = dbConn.execCommandN('SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA')
+
+    dbs = [row[0] for row in dbs]
+    _log.debug('dbs(): dbs = %s', dbs)
+
+    dbData = []
+    for dbName in dbs:
+        if dbName not in _specialDbs:
+            dbData += [_dbDict(dbName)]
+
+    return json.jsonify(results=dbData)
+
+
+@dbService.route('', methods=['POST'])
```

Comment:
GET cannot be used to modify resources. See https://confluence.lsstcorp.org/display/DM/REST+API+General+Guidelines (or any other RESTful introduction).